ext/openssl/openssl.c

```diff
@@ -1063,17 +1063,20 @@ PHP_FUNCTION(openssl_x509_parse)
 	add_assoc_string(return_value, "signatureTypeSN", (char*)OBJ_nid2sn(sig_nid));
 	add_assoc_string(return_value, "signatureTypeLN", (char*)OBJ_nid2ln(sig_nid));
 	add_assoc_long(return_value, "signatureTypeNID", sig_nid);
-	array_init(&subitem);
+
+	int x509_count = X509_PURPOSE_get_count();
+	array_init_size(&subitem, x509_count);
 
 	/* NOTE: the purposes are added as integer keys - the keys match up to the X509_PURPOSE_SSL_XXX defines
 	   in x509v3.h */
-	for (i = 0; i < X509_PURPOSE_get_count(); i++) {
+	for (i = 0; i < x509_count; i++) {
 		int id, purpset;
 		char * pname;
 		X509_PURPOSE * purp;
 		zval subsub;
 
-		array_init(&subsub);
+		array_init_size(&subsub, 3);
+		zend_hash_real_init_packed(Z_ARRVAL(subsub));
```

Comment:
if this gives visible performance improvement, why don't you create array_init variants (e.g. `array_init_packed` and `array_init_size_packed`)?

Reply:
I just might. Not huge differences by any mean but I ran those

![image](https://github.com/user-attachments/assets/2cc8d64d-0613-46cb-bd66-11038221a10b)